src/main/java/dk/kb/storage/storage/DsStorage.java

```diff
@@ -697,6 +718,65 @@ public int updateMTimeForRecord(String recordId) throws Exception {
     }
     
     
+    
+      /**
+       * Create a new entry in the mapping table. The kalturaid can be null and will be updated by a job later.
+       * 
+       * @param mapping The mapping.id must not be null. The kalturaid can be null-
+       * @throws Exception If mapping.id already exists.
+       */
+     public void createNewMapping(MappingDto mappingDto) throws Exception {
+
+        if (mappingDto.getReferenceId() == null) {
+            throw new InvalidArgumentServiceException("ReferenceId must not be null"); // TODO exception enum types, messages?
```

Comment:
You have a TODO here, have you forgotten it or explicitly not implemented it. If not implemented maybe we should create a jira and and to the bottom of the backlog?

Reply:
Copy paste error.